### ISSUE TITLE
chore: use Rollup to bundle the lib

### ISSUE THREAD

**Issue by rolandjitsu** [biosustain/potion-node#27]
Use the following Angular lib examples:
- [angular-quickstart-lib](https://github.com/filipesilva/angular-quickstart-lib)
- [simple-ui-lib](https://github.com/jasonaden/simple-ui-lib)

**Comment by lyschoening** [comment]
How does rollup compare to webpack 2?

**Comment by rolandjitsu** [comment]
We only need a bundler, we transpile using ngc/tsc. We will use Rollup for treeshaking and bundling (which is all it does). Webpack generates more code and it's threeshaking impl. does not do what threeshaking should do (remove dead code) whereas Rollup is very simple to configure and it does not output more than necessary.

You can also read more about the differences in the following:
- http://david-rodenas.com/posts/rollup-vs-webpack-and-tree-shaking
- https://github.com/rollup/rollup/issues/219

I considered using Closure at some point, but it's not an easy task to implement and not very stable with Angular 4 and AOT (though it considerably cuts the bundle size).

Furthermore, I will also add [tslib](https://github.com/Microsoft/tslib) as dependency so we can decrease the size of our bundled code (because the TS compiler adds a lot of helper functions in each file it compiles).

Bottom line, I'd choose Rollup for its simplicity and flexibility. And the Angular team has some good examples on how to setup libs that integrate in the Angular ecosystem.

**Comment by lyschoening** [comment]
@rolandjitsu Ok, thanks for the info. We use webpack as a bundler for our apps now, but I also noticed that it isn't a very good solution for bundling libraries.

**Comment by rolandjitsu** [comment]
@lyschoening No problem. There should be no problems with using this lib (if we use Rollup to create bundles) with Webpack as I also use Angular CLI which uses Webpack to run the Angular apps and some libs I use are bundled using Rollup (see Material 2 for example).

**Comment by rolandjitsu** [comment]
@lyschoening I've been working on implementing this and fixing some bugs after fixing the circular ref. issue in [next](https://github.com/biosustain/potion-node/tree/next) branch. It's published under the `next` tag on NPM or `2.0.0`.

There are some breaking changes though, but only for the way you import depending on what lib you use it with. Basically:
```ts
import {...} from 'potion-client/angular' // for Angular 1 -> import {...} from 'potion-client/angularjs'
import {...} from 'potion-client/@angular' // for Angular 4+ -> import {...} from 'potion-client'
import {...} from 'potion-client' // for Fetch -> import {...} from 'potion-client/fetch'
```

But we may consider publishing each impl. as a different package, the same way Angular 2+ does (e.g. `@potion/angularjs`, `@potion/angular`, etc.), but still keep the whole code in the same repository (monolith repo).

**Comment by rolandjitsu** [comment]
Close with https://github.com/biosustain/potion-node/commit/a50efff080fcf07a011b0de6783fa99e13e85289.
